src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/LSPSemanticTokensFileViewProviderHelper.java

```diff
@@ -33,7 +33,33 @@
  */
 public class LSPSemanticTokensFileViewProviderHelper implements LSPSemanticTokensContainer {
 
+    /**
+     * Simple short-duration thread-local cache for the file view provider's PSI file to minimize read-through during
+     * burst reads of information from the file view provider.
+     */
+    private static final class CachedPsiFile {
```

Comment:
This basically implements the recommendation [here](https://github.com/redhat-developer/lsp4ij/discussions/899#discussioncomment-12640287). The main differences are that an invalid cached PSI file will result in a cache read-through, and the start/duration times are measured in ms directly instead of ns which are then converted to ms. The cache is retained for a 1 sec duration, but that can be tuned if desired. Generally that should result in a good cache hit ratio for burst reads while being imperceptible if/when config changes.